homeassistant/util/logging.py

```diff
@@ -66,13 +71,26 @@ def handle(self, record: logging.LogRecord) -> None:
             module_name,
             module_count,
         )
-        self._warned_modules.add(module_name)
+        self._module_log_count_skip_flags[module_name] = True
 
     def _reset_counters(self, time_sec: float) -> None:
         _LOGGER.debug("Resetting log counters")
         self._last_reset = time_sec
         self._log_counts = defaultdict(int)
 
+    def _check_skip_flag(self, module_name: str) -> bool:
+        skip_flag: bool | None = self._module_log_count_skip_flags.get(module_name)
+        if skip_flag:
+            return True
+
+        if skip_flag is None and any(
+            module_name.startswith(prefix) for prefix in self.EXCLUDED_LOG_COUNT_MODULES
+        ):
+            self._module_log_count_skip_flags[module_name] = True
+            return True
```

Comment:
Can we avoid doing a linear search with a genexpr on each log message?

Reply:
There's some caching happening to avoid it on lines 82-84, but the cache needs to store the result of the matching, not only if the matching is a hit.